api/src/main/java/org/openmrs/api/context/Context.java

```diff
@@ -1427,4 +1428,47 @@ public static void setUseSystemClassLoader(boolean useSystemClassLoader) {
 	public static boolean isUseSystemClassLoader() {
 		return getServiceContext().isUseSystemClassLoader();
 	}
+	
+	/**
+	 * Evicts the entity data for a particular entity "instance".
+	 *
+	 * @param sessionFactory
+	 * @since 2.6.0
+	 */
+	public static void evictSingleEntity(SessionFactory sessionFactory, Class<?> entityClass, Integer id) {
+		if((id == null) || (entityClass == null)) {
+			throw new IllegalArgumentException();
+		}
+		log.debug("Clearing DB cache for entity: {} with id: {}", entityClass, id);
+		sessionFactory.getCache().evictEntity(entityClass, id);
+		sessionFactory.getCache().evictCollectionRegions();
+		sessionFactory.getCache().evictQueryRegions();
+	}
+
+	/**
+	 * Evicts all entity data from the given region (i.e. for all entities of type).
+	 *
+	 * @param sessionFactory
+	 * @since 2.6.0
+	 */
+	public static void evictAllEntities(SessionFactory sessionFactory, Class<?> entityClass) {
+		if(entityClass == null){
```

Comment:
change to this `if(StringUtils.isNotEmpty(entityClass)){}`

as well

generally avoid if statements which resolve to null

Reply:
I guess that method is only applicable for strings. When I changed it the compilation failed.